### PR TITLE
Adds GET request for all articles

### PR DESCRIPTION
**Pull request by nkytruong**
I've added an endpoint for GET /api/articles

- New controller
- New model
- New test

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -1,80 +1,101 @@
...
-                    author: expect.any(String),
-                    body: expect.any(String),
-                    created_at: expect.any(String),
-                    votes: expect.any(Number),
-                    article_img_url: expect.any(String),
-                }
-            )
-            console.log(body)
+        .then(({body: {articles}}) => {
+            expect(articles.length = 13)
```
Comment (MKD01) [inline __tests__/index.test.js:85]: This expect statement isnt finished but is very important so I am glad it was included.

File: __tests__/index.test.js [hunk 2]
```diff
@@ -1,80 +1,101 @@
...
+                        author: expect.any(String),
+                        title: expect.any(String),
+                        article_id: expect.any(Number),
+                        topic: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        article_img_url: expect.any(String),
+                        comment_count: expect.any(Number),
+                    })
+
```
Comment (MKD01) [inline __tests__/index.test.js:97]: Really good assertions

File: db/models/articles-models.js [hunk 3]
```diff
@@ -1,22 +1,39 @@
...
-    if(notANumber){
-        return Promise.reject({status: 400, msg: "Bad Request"})
-    } else {
-        return db.query(queryStr, queryValues).then((result) => {
-            return result.rows
-        })
+  const notANumber = /[a-zA-Z]/.test(article_id);
+  if (notANumber) {
+    return Promise.reject({ status: 400, msg: "Bad Request" });
+  } else {
```
Comment (MKD01) [inline db/models/articles-models.js:15]: This can and should be handled by an error thrown by psql

**Review by MKD01 — COMMENTED** [review]
There is an expect statement that should be updated, but overall really good work. Once you have made the changes please merge them and you dont need to send another pr for this either.

File: __tests__/index.test.js (hunk 1 above)
Comment (nkytruong) [reply __tests__/index.test.js:85]: Do I need to find a way to make it dynamic? So that if new articles are added, it will also include the new article?

Comment (MKD01) [reply __tests__/index.test.js:85]: This expect statement isnt checking anything, by using a single = you are attempting to assign or re-assign something. Here it looks like you are trying to re-assign the length of an array which isnt checking its length. I would recommending looking into the .toBeGreaterThan matcher for this.

Comment (nkytruong) [reply __tests__/index.test.js:85]: Oh I see, if I use .toBeGreaterThan would I also need to use .toBeLessThan? So if I was to do .toBeGreaterThan(12) and .toBeLessThan(14) as I am testing if the length is equal to 13?
Alternatively, if I was to use strict equality === would that also work?

**Pull request merged**